src/Microsoft.Health.Fhir.Api/Controllers/ExportController.cs

```diff
@@ -94,6 +128,36 @@ public IActionResult ExportResourceTypeById(string type, string id)
             return CheckIfExportIsEnabledAndRespond();
         }
 
+        [HttpGet]
+        [Route(KnownRoutes.ExportStatusById, Name = RouteNames.GetExportStatusById)]
+        [AuditEventType(AuditEventSubType.Export)]
+        public async Task<IActionResult> GetExportStatusById(string id)
+        {
+            var getExportResult = await _mediator.GetExportStatusAsync(_fhirRequestContextAccessor.FhirRequestContext.Uri, id);
+
+            if (!getExportResult.JobExists)
+            {
+                throw new JobNotFoundException(string.Format(Resources.JobNotFoundException, id));
+            }
+
+            // If the job is complete, we need to return 200 along the completed data to the client.
+            // Else we need to return 202.
+            OperationResult operationResult;
```

Comment:
The main idea behind using `OperationResult` instead of `FhirResult` (in ExportController) is that we will never be returning a fhir `Resource` as part of the response. So we want the option to send other types of responses. Whenever we throw exceptions here we depend on OperationOutcomeExceptionFilterAttribute to handle it approrpriately. That always uses a `FhirResult` when creating a response. Practically it does not make any difference since we are just returning an error (OperationOutcome) to the client and it shouldn't matter if we use `FhirResult` or `OperationResult` for that. But it does mean `ExportController` will return different types of action results.

Reply:
Can we follow the existing pattern and extract these to OperationResult.Accepted() and OperationResult.Create?